### PR TITLE
AI Integrity Restorer EMP Rework

### LINKED ISSUE #4982: AI repair console - cannot be repair if damaged whilst repairing AI
**Problem Description**:
What is the problem?

If the AI repair console in the RD's office is damaged from EMPs, Laser shots or anything else. It cannot be repaired as you receive a message saying the AI repair must be completed first (something to that effect).

**What did you expect to happen**: 
Why do you think this is a bug?

AIs get stuck in purgatory and cannot be removed from the console, nor the console repaired to fix the AI.

**What happened instead**:
How is what happened different from what you expected?

AIs being repaired and the repair fails should still be able to be uploaded to a card or the consle repaired with the AI repair being continued afterwards.

**Why is this bad/What are the consequences:**
Why do you think this is an important issue?

An AI player is effectively removed from the round if the console is damaged or EMPed since it cannot be repaired if the AI was being repaired at the time the console was damaged.

**Steps to reproduce the problem**:
The most important section. Review everything you did leading up to causing the issue.

Upload AI into repair console. Set off EMP near console. Try to repair console and receive warning message saying the AI isn't repaired yet.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: Anything else you can tell us.

### PR DESCRIPTION
**Pull request by TullyBurnalot**
Fixes #4982

- Budget cuts have led to hardware downgrading, causing EMPs to destroy any AI stored inside an AI Integrity Restorer;

- AI Integrity Restorers that are broken via explosions can be deconstructed. This, of course, kills the AI

:cl:
tweak: AI Integrity Restorers now destroy any AIs inside if they get EMP'd
tweak: AI Integrity Restorers can be deconstructed if broken via explosions. This kills the AI inside
/:cl:

### CODE REVIEW

**Comment by TheDZD** [comment]
Just use `Fixes #abcd` or `Resolves #abcd` or something similar with any of the words mentioned [here](https://help.github.com/articles/closing-issues-via-commit-messages/) so issues automatically closes the issue in question. `Fixes/builds on #4982` would not close the issue automatically.

**Comment by TullyBurnalot** [comment]
@TheDZD Done

File: code/game/machinery/computer/aifixer.dm [hunk 1]
```diff
@@ -135,3 +131,10 @@
 			to_chat(user, "<span class='boldannounce'>ERROR</span>: Reconstruction in progress.")
 		else if(!occupant)
 			to_chat(user, "<span class='boldannounce'>ERROR</span>: Unable to locate artificial intelligence.")
+
+/obj/machinery/computer/aifixer/emp_act()
+	if(occupant)
+		restoring = 0
+		occupant.adjustBruteLoss(200)
```
Comment (Regen1) [inline code/game/machinery/computer/aifixer.dm]: It should be oxyloss

Comment (TullyBurnalot) [inline code/game/machinery/computer/aifixer.dm]: Bruteloss works just as fine. Is there any functional difference?

**Comment by Fox-McCloud** [comment]
IMO it should permakill the AI if it gets EMP'd while its in there.

If you EMP a cloner while it's doing its thing, it perma-kills the person; this is just the AI's cloner, effectively.

File: code/game/machinery/computer/aifixer.dm [hunk 2]
```diff
@@ -7,17 +7,11 @@
 	req_access = list(access_captain, access_robotics, access_heads)
 	var/mob/living/silicon/ai/occupant = null
 	var/active = 0
+	var/restoring = 0
 
 	light_color = LIGHT_COLOR_PURPLE
 
 /obj/machinery/computer/aifixer/attackby(I as obj, user as mob, params)
```
Comment (Fox-McCloud) [inline code/game/machinery/computer/aifixer.dm]: if this just runs parent, then his entire proc override is unnecessary.

**Comment by TullyBurnalot** [comment]
@Fox-McCloud My main issue with just outright deleting the AI entirely is that organic cloning requires two different machines to be both broken, the pod and the records computer.

Meanwhile, the AI has **one** possible system that it can use to be "cloned", and its loss has a far heavier impact than, say, a random crewmember.

Plus, there are simpler, more cost effective ways of removing an AI from the round. Such as turning off their wireless and radio, then either keeping it on you, spacing it, or, if you're feeling particularly evil, cremating it, the latter of which requires only breaking **one** rarely checked door as opposed to breaking into a Head of Staff office.

File: code/game/machinery/computer/aifixer.dm (hunk 1 above)
Comment (Fox-McCloud) [inline code/game/machinery/computer/aifixer.dm]: Tully is probably right here because well, when you EMP an AI it takes brute damage by default.

**Comment by TullyBurnalot** [comment]
@Fox-McCloud Alternatively, I could keep the last two parts, and remove the BREAK immunity, so as long as RNG is on your side, you'll "just" be reset to 0, but you'll still get deleted permanently if the EMP happens to break the console.

File: code/game/machinery/computer/aifixer.dm [hunk 3]
```diff
@@ -135,3 +128,13 @@
 			to_chat(user, "<span class='boldannounce'>ERROR</span>: Reconstruction in progress.")
 		else if(!occupant)
 			to_chat(user, "<span class='boldannounce'>ERROR</span>: Unable to locate artificial intelligence.")
+
+/obj/machinery/computer/aifixer/emp_act()
+	if(occupant)
+		occupant.ghostize()
+		qdel(occupant)
```
Comment (Fox-McCloud) [inline code/game/machinery/computer/aifixer.dm]: may want to null out occupant here, just in case.

Comment (TullyBurnalot) [inline code/game/machinery/computer/aifixer.dm]: Added.

File: code/game/machinery/computer/aifixer.dm [hunk 4]
```diff
@@ -135,3 +128,12 @@
 			to_chat(user, "<span class='boldannounce'>ERROR</span>: Reconstruction in progress.")
 		else if(!occupant)
 			to_chat(user, "<span class='boldannounce'>ERROR</span>: Unable to locate artificial intelligence.")
+
+/obj/machinery/computer/aifixer/emp_act()
+	if(occupant)
+		occupant.ghostize()
+		qdel(occupant)
+		occupant = null
+		return
```
Comment (Fox-McCloud) [inline code/game/machinery/computer/aifixer.dm]: trailing return

File: code/game/machinery/computer/aifixer.dm [hunk 5]
```diff
@@ -11,13 +11,6 @@
 	light_color = LIGHT_COLOR_PURPLE
 
 /obj/machinery/computer/aifixer/attackby(I as obj, user as mob, params)
-	if(occupant && istype(I, /obj/item/weapon/screwdriver))
-		if(stat & (NOPOWER|BROKEN))
-			to_chat(user, "<span class='warning'>The screws on [name]'s screen won't budge.</span>")
-		else
-			to_chat(user, "<span class='warning'>The screws on [name]'s screen won't budge and it emits a warning beep.</span>")
-		return
-	else
```
Comment (Fox-McCloud) [inline code/game/machinery/computer/aifixer.dm]: this should probably stay, or else you can just kill the AI, permanently by merely unscrewing the console.

Comment (TullyBurnalot) [inline code/game/machinery/computer/aifixer.dm]: ... acceptable trade-off? *shrug

File: code/game/machinery/computer/aifixer.dm [hunk 6]
```diff
@@ -12,11 +12,14 @@
 
 /obj/machinery/computer/aifixer/attackby(I as obj, user as mob, params)
 	if(occupant && istype(I, /obj/item/weapon/screwdriver))
-		if(stat & (NOPOWER|BROKEN))
+		if(stat & BROKEN)
+			..()
+		if(stat & NOPOWER)
 			to_chat(user, "<span class='warning'>The screws on [name]'s screen won't budge.</span>")
+			return
```
Comment (Fox-McCloud) [inline code/game/machinery/computer/aifixer.dm]: Trailing returnsssss

File: code/game/machinery/computer/aifixer.dm [hunk 7]
```diff
@@ -12,11 +12,14 @@
...
+		if(stat & BROKEN)
+			..()
+		if(stat & NOPOWER)
 			to_chat(user, "<span class='warning'>The screws on [name]'s screen won't budge.</span>")
+			return
 		else
-			to_chat(user, "<span class='warning'>The screws on [name]'s screen won't budge and it emits a warning beep.</span>")
-		return
+			to_chat(user, "<span class='warning'>The screws on [name]'s screen won't budge and it emits a warning beep!.</span>")
+			return
```
Comment (Fox-McCloud) [inline code/game/machinery/computer/aifixer.dm]: Trailing returnsssss

**Comment by Fox-McCloud** [comment]
I just noticed something....

this needs
```
/obj/machinery/computer/aifixer/Destroy()
	if(occupant)
		occupant.ghostize()
		qdel(occupant)
		occupant = null
```

**Comment by Fox-McCloud** [comment]
You going to work on this more or not?

**Pull request merged**